source/common/common/posix/thread_impl.cc

```diff
@@ -140,15 +171,20 @@ int PosixThreadFactory::createPthread(ThreadHandle* thread_handle) {
   return pthread_create(
       &thread_handle->handle(), nullptr,
       [](void* arg) -> void* {
-        static_cast<ThreadHandle*>(arg)->routine()();
+        ThreadHandle* handle = static_cast<ThreadHandle*>(arg);
+        if (handle->priority()) {
+          setThreadPriority(getCurrentThreadId(), *handle->priority());
+        }
+        handle->routine()();
         return nullptr;
       },
       reinterpret_cast<void*>(thread_handle));
 }
 
 PosixThreadPtr PosixThreadFactory::createThread(std::function<void()> thread_routine,
                                                 OptionsOptConstRef options, bool crash_on_failure) {
-  auto thread_handle = new ThreadHandle(thread_routine);
+  auto thread_handle =
+      new ThreadHandle(thread_routine, options ? options->priority_ : absl::nullopt);
```

Comment:
Can we just unconditionally pass in `options->priority_`? (If it's false I think it'll be nullopt, right?)

Reply:
OH! `options` itself is optional! HAH. Right. Sorry. I confused `options` for `options->priority`! Sorry about that.

Reply:
I feel like there's a tongue twister buried somewhere in here.... :)